core/src/main/scala/akka/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -69,14 +69,22 @@ class JournalQueries(
     JournalTable.filter(_.persistenceId === persistenceId).filter(_.sequenceNumber <= toSequenceNr).delete
   }
 
-  def markAsDeleted(persistenceId: String, highestMarkedSequenceNr: Long) =
+  private[akka] def markAsDeleted(persistenceId: String, highestMarkedSequenceNr: Long) =
     JournalTable
       .filter(_.persistenceId === persistenceId)
       .filter(_.sequenceNumber === highestMarkedSequenceNr)
       .filter(_.deleted === false)
       .map(_.deleted)
       .update(true)
 
+  def markJournalMessagesAsDeleted(persistenceId: String, maxSequenceNr: Long) =
```

Comment:
Mark as deprecated?

Reply:
With what message? We are not offering an alternative to it.

If we deprecate it and someone is using, it more like: we will remove it and you have to give up from whatever you are doing with this method because there is no alternative. 

But I might be overthinking it. This is mostly internal API.

Reply:
I added this:

`@deprecated(message = "Intended to be internal API", since = "5.4.2")`